hs-bindgen/src/HsBindgen/C/Fold/Type.hs

```diff
@@ -124,6 +127,8 @@ mkStructField unit current = do
       fieldOffset <- fromIntegral <$> clang_Cursor_getOffsetOfField current
       fieldName   <- CName <$> clang_getCursorDisplayName current
 
+      unless (fieldOffset `mod` 8 == 0) $ error "bit-fields not supported yet"
```

Comment:
`fail`. Don't use `error` in `IO`.

Reply:
@edsko requested that I use `error` for such assertions, only using `fail` for things that may be handled.  (I put similar assertions in the documentation reification, also running in `IO`.)